pw-pbkdf2.c

```diff
@@ -229,6 +235,81 @@ static int pbkdf2_encrypt(
 						  dk.bv_len, iteration,
 						  salt.bv_len, (const uint8_t *) salt.bv_val,
 						  dk.bv_len, dk_value);
+#elif HAVE_MOZNSS
+	SECStatus rv;
+	rv = NSS_NoDB_Init(".");
+	if (rv != SECSuccess)
+	{
+		fprintf(stderr, "NSS initialization failed (err %d)\n",
+		PR_GetError());
+	}
+	PK11SlotInfo *slot = NULL;
+	SECAlgorithmID * algid = NULL;
+	PRArenaPool *arena_001;
+	PRArenaPool *arena_002;
+	PRArenaPool *arena_003;
+	arena_001 = PORT_NewArena(DER_DEFAULT_CHUNKSIZE);
+	arena_002 = PORT_NewArena(DER_DEFAULT_CHUNKSIZE);
+	arena_003 = PORT_NewArena(DER_DEFAULT_CHUNKSIZE);
+	SECItem salt_moz={ siBuffer, NULL, 0};
+	SECItem pwitem={ siBuffer, NULL, 0};
+	slot=PK11_GetBestSlot(SEC_OID_PKCS5_PBKDF2, NULL);
+	if (slot == NULL)
+	{
+		printf("error initializing slot");
```

Comment:
output to use stderr and
return LUTIL_PASSWD_ERR;

Reply:
Done